relay/src/cli.rs

```diff
@@ -314,6 +319,57 @@ pub fn init_config<P: AsRef<Path>>(config_path: P, _matches: &ArgMatches) -> Res
     Ok(())
 }
 
+/// Manages the on-disk spool file.
+pub fn manage_spool(config: &Config, matches: &ArgMatches) -> Result<()> {
+    if let Some(matches) = matches.subcommand_matches("clear") {
+        let path = match matches.get_one::<String>("path") {
+            Some(path) => {
+                let Ok(path) = PathBuf::from_str(path) else {
+                    bail!("Can't find the provided file path.")
+                };
+                path
+            }
+            None => {
+                let Some(path) = config.spool_envelopes_path() else {
+                    bail!("Config file does not contain the path to the spool file.")
+                };
+                path
+            }
+        };
+
+        if !path.is_file() {
+            bail!("Could not find provided file: {}", path.to_string_lossy());
+        }
+
+        let force = matches.get_flag("force");
+
+        if !force {
+            let stdin = std::io::stdin();
+            eprintln!(
+                "Are you sure you want to clear up on-disk spooled data in {} (yes/no): ",
+                path.to_string_lossy()
+            );
+            for line in stdin.lock().lines().map_while(Result::ok) {
+                match line.trim().to_lowercase().as_str() {
+                    "yes" => break,
+                    "no" => bail!("Canceling the cleaning of the spooled data."),
+                    _ => eprintln!("Accepting only YES or NO: "),
+                }
+            }
+        }
+
+        relay_log::info!("Clearing the spool file: {}", path.to_string_lossy());
+
+        spool_utils::truncate(path)?;
+
+        relay_log::info!("On-disk spool emptied.")
+    } else {
+        unreachable!();
```

Comment:
Might as well return an error, via `bail` imo

Reply:
Followed the pattern like other functions were done, but erroring here makes sense, will change.

Reply:
This also actually never happens that's why this is really unreachable , the `clap` does all the validation and checks before.